operator/src/main/java/io/strimzi/kafka/access/KafkaAccessReconciler.java

```diff
@@ -208,7 +212,7 @@ public ErrorStatusUpdateControl<KafkaAccess> updateErrorStatus(KafkaAccess kafka
         }
         status.setReadyCondition(false, e.getMessage(), reason);
 
-        return ErrorStatusUpdateControl.updateStatus(kafkaAccess);
+        return ErrorStatusUpdateControl.patchStatus(kafkaAccess);
```

Comment:
Similarly here we will need to create a new KafkaAccess and KafkaAccessStatus object.